code/go/0chain.net/smartcontract/dbs/event/user_snapshot.go

```diff
@@ -0,0 +1,32 @@
+package event
+
+import (
+	"time"
+
+	"gorm.io/gorm/clause"
+)
+
+
+type UserSnapshot struct {
+	UserID          string `json:"user_id" gorm:"uniqueIndex"`
+	Round           int64  `json:"round"`
+	CollectedReward int64  `json:"collected_reward"`
+	TotalStake      int64  `json:"total_stake"`
+	ReadPoolTotal   int64  `json:"read_pool_total"`
+	WritePoolTotal  int64  `json:"write_pool_total"`
+	PayedFees       int64  `json:"payed_fees"`
+	CreatedAt       time.Time
+	UpdatedAt       time.Time
+}
+
+func (edb *EventDb) GetUserSnapshotsByIds(ids []string) (snapshots []UserSnapshot, err error) {
+	err = edb.Store.Get().Where("user_id in (?)", ids).Find(&snapshots).Error
```

Comment:
use temp table for it, it's preferable then `in ?` since every different set of parameters generate different query and different plan as well, it is all cached in db and wastes resources